src/overlay005/field_control.c

```diff
@@ -0,0 +1,1092 @@
+            if (sub_0205F588(fieldSystem->playerAvatar) == TRUE) {
+                sub_0205F5E4(fieldSystem->playerAvatar, PlayerAvatar_GetDir(fieldSystem->playerAvatar));
+            }
+
+            sub_0203E880(fieldSystem, sub_02062960(object), object);
+            return TRUE;
+        }
+    }
+
+    if (input->interact) {
+        sub_0205B2D4(fieldSystem);
+        return TRUE;
+    }
+
+    if (input->menu) {
+        Sound_PlayEffect(SEQ_SE_DP_WIN_OPEN);
+        sub_0203AABC(fieldSystem);
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+static int Field_CheckTrainerInfo (void)
+{
+    int netId;
+    TrainerInfo *info;
+
+    int ret = 0;
+
+    for (netId = 1; netId < 5; netId++) {
+        info = CommInfo_TrainerInfo(netId);
+
+        if (info != NULL) {
+            ret++;
+        }
+    }
+
+    return ret >= 1;
+}
+
+BOOL FieldInput_Process_UnionRoom (const FieldInput *input, FieldSystem *fieldSystem)
+{
+    if (sub_020318EC() & 0xfe) {
+        if (Field_CheckTrainerInfo() && CommSys_ConnectedCount() > 1) {
+            sub_0203E880(fieldSystem, 5, NULL);
+            return TRUE;
+        }
+
+        return FALSE;
+    }
+
+    if (input->interact) {
+        MapObject *object;
+
+        if (sub_0203CA40(fieldSystem, &object) == TRUE) {
+            if (sub_0205F588(fieldSystem->playerAvatar) == TRUE) {
+                sub_0205F5E4(fieldSystem->playerAvatar, PlayerAvatar_GetDir(fieldSystem->playerAvatar));
+            }
+
+            sub_02036B84();
+            sub_0203E880(fieldSystem, sub_02062960(object), object);
+
+            return TRUE;
+        }
+    }
+
+    if (input->movement && sub_0205DEE4(Field_CurrentTileBehavior(fieldSystem))) {
+        sub_020545EC(fieldSystem);
+        return TRUE;
+    }
+
+    if (input->menu && CommSys_ConnectedCount() <= 1) {
+        Sound_PlayEffect(SEQ_SE_DP_WIN_OPEN);
+        sub_0203AA78(fieldSystem);
+        sub_0205BEA8(4);
+        sub_02036BA0();
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+int FieldInput_Process_BattleTower (const FieldInput *input, FieldSystem *fieldSystem)
+{
+    if (input->unused5 == FALSE && sub_0203F5C0(fieldSystem, 1) == TRUE) {
+        return TRUE;
+    }
+
+    if (input->interact) {
+        MapObject *object;
+
+        if (sub_0203CA40(fieldSystem, &object) == TRUE) {
+            if (sub_0205F588(fieldSystem->playerAvatar) == TRUE) {
+                sub_0205F5E4(fieldSystem->playerAvatar, PlayerAvatar_GetDir(fieldSystem->playerAvatar));
+            }
+            
+            if (sub_02062950(object) != 0x9) {
+                sub_0203E880(fieldSystem, sub_02062960(object), object);
+            } else {
+                sub_0203E880(fieldSystem, 0, object);
+            }
+
+            return TRUE;
+        }
+
+        int bgEvent = sub_0203CA6C(fieldSystem, (void *)sub_0203A440(fieldSystem), sub_0203A448(fieldSystem));
```

Comment:
Why name this `bgEvent`?

Reply:
Bump

Reply:
Yeah this name is just not good. I wanted to distinguish it from the event potentially resulting from tile behavior (the one right below), but I think I'll need to review `sub_0203CA6C()` and the file it belongs to a lot more before having a good name for it. I'll revert to the original name for now and come up with a good one when I flesh out more here.